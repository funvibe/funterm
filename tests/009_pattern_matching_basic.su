for num = 1,3 do
  match num {
    0 -> lua.print("zero"),
    1 -> lua.print("one"),
    2 -> lua.print("two"),
    _ -> lua.print("No match")
  }
end

lua.data = ["", "one", "two"]
for i = 0,2 do
  match lua.data[i] {
    "zero" -> lua.print("zero"),
    "one"  -> lua.print("one"),
    "two"  -> lua.print("two"),
    _      -> lua.print("No match")
  }
end