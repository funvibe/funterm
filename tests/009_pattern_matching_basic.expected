one
two
No match
No match
one
two